lib/xlsx/parts/shared_strings.rb

```diff
@@ -4,17 +4,17 @@ class SharedStrings < OpenXml::Part
       attr_reader :strings
 
       def initialize
-        @strings = []
+        @strings = Hash.new { |hash, key| hash[key] = hash.length }
       end
 
       def reference_of(string)
-        Xlsx.index!(strings, string)
+        strings[string]
       end
 
       def to_xml
         build_standalone_xml do |xml|
           xml.sst(xmlns: "http://schemas.openxmlformats.org/spreadsheetml/2006/main", uniqueCount: strings.length) do
-            strings.each do |string|
+            strings.each do |string, i|
```

Comment:
Unused block argument - `i`. If it's necessary, use `_` or `_i` as an argument name to indicate that it won't be used.